File: functions/libsodium/variants/go/1_b64_byte_to_urlsafe_char_0.go
package main

func b64ByteToURLSafeChar(x uint32) int32 {
    if x < 26 {
        return int32(x + 'A')
    } else if x < 52 {
        return int32(x - 26 + 'a')
    } else if x < 62 {
        return int32(x - 52 + '0')
    } else if x == 62 {
        return '-'
    } else if x == 63 {
        return '_'
    }
    return -1
}


func main(){
    b64ByteToURLSafeChar(0)
}

File: functions/libsodium/variants/c/1_b64_byte_to_urlsafe_char_0.c
#include <stdint.h>

static int b64_byte_to_urlsafe_char(unsigned int x)
{
    if (x < 26) return x + 'A';
    if (x < 52) return x - 26 + 'a';
    if (x < 62) return x - 52 + '0';
    if (x == 62) return '-';
    if (x == 63) return '_';
    return -1; // Should never reach here
}


int main(){
    b64_byte_to_urlsafe_char(0);
}
